DoSomething/gladiator
# Transforming like a boss

#### What's this PR do?
This PR implements the use of the Fractal library to transform API responses and standardize what they should be across the Gladiator API. It also updates the `/api/v1/users` POST endpoint to take into account the new DB schema and checks against the Contest -> WaitingRoom to see if a user already exits in the room before adding them.

This also removes the `/api/v1/waitingrooms` endpoint which we are no longer using with the new DB schema, and replaced it with `/api/v1/contests`.

#### How should this be manually tested?
Pull down branch, `composer dump-autoload` try hitting `/api/v1/users` with a POST request based on the [API doc](https://github.com/DoSomething/gladiator/blob/master/documentation/endpoints/users.md#create-a-user) and you should see a JSON response with the user info (keep in mind all values apart from `id` will be null cause that shiz is coming in a different PR with caching the user data!)

```javascript
{
  "data": {
    "id": "697efkhad04858403adnas43",  // MongoDB id
    "first_name": null,
    "last_name": null,
    "email": null,
    "mobile": null,
    "signup": null,
    "reportback": null
  }
}
```

#### Any background context you want to provide?
Waiting on Contest CRUD #59 before I can format and return an updated response for Phoenix when asking about contest availability.

#### What are the relevant tickets?
Fixes #48 


---
@DFurnes @angaither 

cc: @sbsmith86 @deadlybutter
